compute/encryption.go

```diff
@@ -0,0 +1,177 @@
+					CreateOption: compute.Empty,
+				},
+				DiskSizeGB: to.Int32Ptr(64),
+			},
+		})
+	if err != nil {
+		return disk, fmt.Errorf("cannot create disk: %v", err)
+	}
+
+	err = future.WaitForCompletion(ctx, disksClient.Client)
+	if err != nil {
+		return disk, fmt.Errorf("cannot get the disk create or update future response: %v", err)
+	}
+
+	return future.Result(disksClient)
+}
+
+func GetDisk(ctx context.Context, diskName string) (disk compute.Disk, err error) {
+	disksClient := getDisksClient()
+	return disksClient.Get(ctx, helpers.ResourceGroupName(), diskName)
+}
+
+func CreateVMWithManagedDisk(ctx context.Context, nicName, diskName, storageAccountName, vmName string) (vm compute.VirtualMachine, err error) {
+	nic, _ := network.GetNic(ctx, nicName)
+	storageAccount, _ := storage.GetStorageAccount(ctx, storageAccountName)
+	disk, _ := GetDisk(ctx, diskName)
+
+	var storageURI *string
+	if storageAccount.PrimaryEndpoints != nil {
+		storageURI = (*storageAccount.PrimaryEndpoints).Blob
+	} else {
+		err = errors.New("No storage endpoint found")
+		return
+	}
+
+	vmClient := getVMClient()
+	future, err := vmClient.CreateOrUpdate(
+		ctx,
+		helpers.ResourceGroupName(),
+		vmName, compute.VirtualMachine{
+			Location: to.StringPtr(helpers.Location()),
+			VirtualMachineProperties: &compute.VirtualMachineProperties{
+				DiagnosticsProfile: &compute.DiagnosticsProfile{
+					BootDiagnostics: &compute.BootDiagnostics{
+						Enabled:    to.BoolPtr(true),
+						StorageURI: storageURI,
+					},
+				},
+				HardwareProfile: &compute.HardwareProfile{
+					VMSize: compute.StandardDS2V2,
+				},
+				NetworkProfile: &compute.NetworkProfile{
+					NetworkInterfaces: &[]compute.NetworkInterfaceReference{
+						{
+							ID: nic.ID,
+							NetworkInterfaceReferenceProperties: &compute.NetworkInterfaceReferenceProperties{
+								Primary: to.BoolPtr(true),
+							},
+						},
+					},
+				},
+				OsProfile: &compute.OSProfile{
+					ComputerName:  to.StringPtr(vmName),
+					AdminUsername: to.StringPtr(username),
+					AdminPassword: to.StringPtr(password),
+					LinuxConfiguration: &compute.LinuxConfiguration{
+						DisablePasswordAuthentication: to.BoolPtr(false),
+					},
+				},
+				StorageProfile: &compute.StorageProfile{
+					ImageReference: &compute.ImageReference{
+						Publisher: to.StringPtr(publisher),
+						Offer:     to.StringPtr(offer),
+						Sku:       to.StringPtr(sku),
+						Version:   to.StringPtr("latest"),
+					},
+					OsDisk: &compute.OSDisk{
+						CreateOption: compute.DiskCreateOptionTypesFromImage,
+						DiskSizeGB:   to.Int32Ptr(64),
+					},
+					DataDisks: &[]compute.DataDisk{
+						{
+							CreateOption: compute.DiskCreateOptionTypesAttach,
+							Lun:          to.Int32Ptr(0),
+							ManagedDisk: &compute.ManagedDiskParameters{
+								ID:                 disk.ID,
+								StorageAccountType: compute.StorageAccountTypes(storageAccount.Sku.Name),
+							},
+						},
```

Comment:
Can we use managed disks ehre also ?

Reply:
Managed disks are already in use here when creating the VM. They were created in a previous step. Will update comment.